server/src/main/java/org/elasticsearch/plugins/PluginsService.java

```diff
@@ -295,6 +296,31 @@ static void loadExtensions(Collection<LoadedPlugin> plugins) {
         }
     }
 
+    // package private for testing
+    <T> Iterator<T> providersIterator(Class<T> service, ClassLoader classLoader) {
+        return ServiceLoader.load(service, classLoader).iterator();
+    }
+
+    /**
+     * SPI convenience method that uses the {@link ServiceLoader} JDK class to load various SPI providers
+     * from plugins/modules. For example:
+     *
+     * var pluginHandlers = pluginsService.loadServiceProviders(OperatorHandlerProvider.class);
+     *
+     * @param service A templated service class to look for providers in plugins
+     * @return an immutable {@link List} of discovered providers in the plugins/modules
+     */
+    public <T> List<? extends T> loadServiceProviders(Class<T> service) {
```

Comment:
Can this be split out into a separate PR? It seems like this is most of this PR, but the title would suggest otherwise.

Reply:
Yes, sounds good, I'll take this out from this PR and contribute it back once the operator handler interface is in.